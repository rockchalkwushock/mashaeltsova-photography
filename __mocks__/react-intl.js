/**
 * Many thanks to @stipsan
 * @see https://medium.com/@stipsan/testing-with-jest-15-awesome-tips-and-tricks-42150ec4c262
 */

const Intl = require.requireActual('react-intl')

// Initialize the real provider so that we don't
// need to re-implement any internals
const defaultProps = {
  locale: 'en',
  defaultLocale: 'en'
}
const intlProvider = new Intl.IntlProvider(defaultProps, {})

// The exact same `intl` object the real code receives ;-)
const { intl } = intlProvider.getChildContext()
Intl.injectIntl = Node => props => <Node {...props} intl={intl} />

// Override components by implementing the real components
//  providing them the context they need in order to function
const {
  IntlProvider,
  FormattedDate,
  FormattedTime,
  FormattedRelative,
  FormattedNumber,
  FormattedPlural,
  FormattedMessage,
  FormattedHTMLMessage
} = Intl
Intl.FormattedDate = props => (
  <IntlProvider {...defaultProps}>
    <FormattedDate {...props} />
  </IntlProvider>
)
Intl.FormattedTime = props => (
  <IntlProvider {...defaultProps}>
    <FormattedTime {...props} />
  </IntlProvider>
)
Intl.FormattedRelative = props => (
  <IntlProvider {...defaultProps}>
    <FormattedRelative {...props} />
  </IntlProvider>
)
Intl.FormattedNumber = props => (
  <IntlProvider {...defaultProps}>
    <FormattedNumber {...props} />
  </IntlProvider>
)
Intl.FormattedPlural = props => (
  <IntlProvider {...defaultProps}>
    <FormattedPlural {...props} />
  </IntlProvider>
)
Intl.FormattedMessage = props => (
  <IntlProvider {...defaultProps}>
    <FormattedMessage {...props} />
  </IntlProvider>
)
Intl.FormattedHTMLMessage = props => (
  <IntlProvider {...defaultProps}>
    <FormattedHTMLMessage {...props} />
  </IntlProvider>
)

// Set displayName so that snapshots don't use "Unknown" as component name
Intl.FormattedDate.displayName = 'FormattedDate'
Intl.FormattedTime.displayName = 'FormattedTime'
Intl.FormattedRelative.displayName = 'FormattedRelative'
Intl.FormattedNumber.displayName = 'FormattedNumber'
Intl.FormattedPlural.displayName = 'FormattedPlural'
Intl.FormattedMessage.displayName = 'FormattedMessage'
Intl.FormattedHTMLMessage.displayName = 'FormattedHTMLMessage'

// Special hook for tests, real package does not export this
Intl.intl = intl

module.exports = Intl
