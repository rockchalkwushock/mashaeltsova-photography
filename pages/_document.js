import Document, { Head, Main, NextScript } from 'next/document'
import { loadGetInitialProps } from 'next/dist/lib/utils'

// Polyfill Intl API for older browsers
// https://cdn.polyfill.io/v2/polyfill.min.js?features=Intl.~locale.${locale}

// The document (which is SSR-only) needs to be customized to expose the locale
// data for the user's locale for React Intl to work in the browser.
export default class CustomDocument extends Document {
  static async getInitialProps(ctx) {
    const { locale, localeDataScript } = ctx.req
    const props = await loadGetInitialProps(Document, ctx)
    return {
      ...props,
      locale,
      localeDataScript
    }
  }

  render() {
    const { locale, localeDataScript } = this.props
    return (
      <html prefix="og: http://ogp.me/ns#">
        <Head />
        <body>
          <Main />
          <script
            src={`https://cdn.polyfill.io/v2/polyfill.min.js?features=Intl.~locale.${locale}`}
          />
          <script
            dangerouslySetInnerHTML={{
              __html: localeDataScript
            }}
          />
          <NextScript />
        </body>
      </html>
    )
  }
}
