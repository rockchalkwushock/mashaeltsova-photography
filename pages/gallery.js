import { Component } from 'react'
import Head from 'next/head'
import PropTypes from 'prop-types'
import NProgress from 'nprogress'
import { loadGetInitialProps } from 'next/dist/lib/utils'

import { A, Grid, Layout, Photo, SubNav, Thumb } from '../components'
import { fetchData, translate, withIntl } from '../lib'

export class Gallery extends Component {
  static propTypes = {
    photos: PropTypes.array.isRequired,
    intl: PropTypes.object.isRequired,
    locale: PropTypes.string.isRequired,
    messages: PropTypes.object.isRequired,
    url: PropTypes.shape({
      pathname: PropTypes.string.isRequired
    })
  }
  static async getInitialProps(ctx) {
    try {
      const props = await loadGetInitialProps(ctx)
      const { ids } = await fetchData('family')
      return { photos: ids, ...props }
    } catch (err) {
      throw err
    }
  }
  state = {
    currentView: 'family',
    error: false,
    image: null,
    images: null,
    message: '',
    modalPhoto: false
  }
  dismissModal = () => {
    this.setState(state => ({
      ...state,
      image: null,
      modalPhoto: false
    }))
  }
  handleOnClick = async e => {
    const target = translate(e.currentTarget.name)
    try {
      NProgress.start()
      const { error, ids } = await fetchData(target)
      if (error) {
        this.setState(state => ({
          ...state,
          currentView: target,
          error: true,
          image: null,
          images: ids,
          message: error,
          modalPhoto: false
        }))
        NProgress.done()
      }
      this.setState(state => ({
        ...state,
        currentView: target,
        image: null,
        images: ids,
        modalPhoto: false
      }))
      NProgress.done()
    } catch (err) {
      throw err
    }
  }
  showPhoto = id => {
    this.setState(state => ({
      ...state,
      image: id,
      modalPhoto: true
    }))
  }
  renderButtons = () =>
    this.props.messages.galleryNav
      .split(',')
      .map(t => (
        <A
          className="galleryButton"
          key={t}
          onClick={this.handleOnClick}
          text={t}
        />
      ))
  renderError = () => <h1>{this.state.message}</h1>
  renderView() {
    const imgs =
      this.state.currentView === 'family'
        ? this.props.photos
        : this.state.images
    if (this.state.modalPhoto) {
      return (
        <Photo
          onDismiss={() => this.dismissModal()}
          publicId={this.state.image}
        />
      )
    }
    return imgs.map(id => (
      <Thumb
        key={id}
        onClick={() => this.showPhoto(id)}
        publicId={id}
        view={this.state.currentView}
      />
    ))
  }
  render() {
    return (
      <Layout messages={this.props.messages} url={this.props.url}>
        <Head>
          <title>Masha Eltsova Photography | Gallery</title>
          {/* SEO */}
          <meta
            name="description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          {/* SCHEMA.ORG FOR GOOGLE */}
          <meta itemProp="name" content="Masha Eltsova Photography | Gallery" />
          <meta
            itemProp="description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          {/* TWITTER */}
          <meta
            name="twitter:title"
            content="Masha Eltsova Photography | Gallery"
          />
          <meta
            name="twitter:description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          <meta
            name="twitter:url"
            content="https://mashaeltsovaphotography.com/gallery"
          />
          {/* FACEBOOK, PINTREST, GOOGLE+ */}
          <meta
            property="og:title"
            content="Masha Eltsova Photography | Gallery"
          />
          <meta
            property="og:description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          <meta
            property="og:site_name"
            content="Masha Eltsova Photography | Gallery"
          />
          <meta
            property="og:url"
            content="https://mashaeltsovaphotography.com/gallery"
          />
        </Head>
        <SubNav>{this.renderButtons()}</SubNav>
        <Grid>{this.state.error ? this.renderError() : this.renderView()}</Grid>
      </Layout>
    )
  }
}

export default withIntl(Gallery)
