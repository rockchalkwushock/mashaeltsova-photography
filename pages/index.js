import { Component } from 'react'
import Head from 'next/head'
import PropTypes from 'prop-types'

import { messages, WithCloudinary, WithIntl } from '../lib'
import { Footer, Layout, Section } from '../components'

export class IndexPage extends Component {
  static propTypes = {
    intl: PropTypes.object.isRequired
  }
  manipulateText = () => {
    const translatedText = this.props.intl.formatMessage(
      messages.servicesMessage
    )
    const [message, services] = translatedText.split(': ')
    const list = services
      .split(',')
      .map(service => <li key={service}>{service}</li>)
    return { message, list }
  }
  render() {
    const { intl, url } = this.props
    const { message, list } = this.manipulateText()
    return (
      <Layout pathname={url.pathname}>
        <Head>
          <title>{intl.formatMessage(messages.title)}</title>
          <meta description={intl.formatMessage(messages.description)} />
        </Head>
        <Section header>
          <h1 style={{ textAlign: 'center' }}>Hello World</h1>
        </Section>
        <Section about>{intl.formatMessage(messages.aboutMessage)}</Section>
        <Section services>
          <p>{message}:</p>
          <ul>{list}</ul>
        </Section>
        <Section gallery>Gallery</Section>
        <Section booking>form</Section>
        <Footer intl={intl} />
      </Layout>
    )
  }
}

export default WithIntl(WithCloudinary(IndexPage))
