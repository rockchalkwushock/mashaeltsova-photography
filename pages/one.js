import { Component } from 'react'
import { WithData } from '../lib'
import { FormContainer } from '../containers'

class TestPage extends Component {
  render() {
    return <FormContainer />
  }
}

export default WithData(TestPage)
