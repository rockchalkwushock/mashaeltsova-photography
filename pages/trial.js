import { Image, Transformation } from 'cloudinary-react'

// import { PageWithoutData } from '../lib'

// https://res.cloudinary.com/rockchalkwushock/c_scale,dpr_2.0,f_auto,q_auto:best,w_414/e_overlay,fl_tiled,l_logo_as_watermark,o_40,w_150/mother_daughter_black_white

const TrialPage = () => (
  <Image
    cloudName="rockchalkwushock"
    publicId="girl_in_red_dress"
    responsive
    secure
    use_root_path
  >
    <Transformation
      crop="scale"
      dpr="auto"
      fetch_format="auto"
      quality="auto:best"
      width="auto"
    />
  </Image>
)

export default TrialPage

// http://res.cloudinary.com/rockchalkwushock/image/upload/ar_4:6,c_fill,g_auto/c_scale,f_auto,q_auto,w_auto/mep_svg_logo
// <Image publicId="ar_4:6,c_fill,g_auto/c_scale,f_auto,q_auto,w_auto/mep_svg_logo" />
/**
 * NOTE
 *
 * secure prop = https
 * dpr = device-pixel-ratio
 * use_root_path = strips the `/image/upload/` from the url: https://res.cloudinary.com/rockchalkwushock/c_fit,dpr_auto,f_auto,w_320/mep_jpg_logo.jpg
 * url_suffix="basketball-game-in-college" NOTE: private CDN ONLY!!!
 *
 * NOTE q_auto defaults to q_auto:good I will want to use q_auto:best for the majority of the images.
 * less aggressive algorithm that will compress the file size up until file degradation begins
 *
 * REVIEW I can change the CNAME but likely need a paid account.
 * The url would change from:
 * https://res.cloudinary.com/rockchalkwushock/c_fit,dpr_auto,f_auto,w_320/mep_jpg_logo.jpg
 * to
 * https://mashaeltsovaphotography.com/imgTitle.jpg
 *
 * Responsive Images should be as easy as:
 * `<Transformation dpr='auto' width='auto' crop='scale' />`
 *
 * REVIEW
 * To optimize the images
 * `<Transformation quality='auto' fetch_format='auto' />`
 * Cloudinary will figure out what is the best quality to send and format.
 * This will reduce the cost of the hi-res photo.
 *
 * NOTE https://webspeedtest.cloudinary.com/
 * Use that to test the website when I am ready to go into production to verify
 * any images that can be further optimized.
 *
 * BUG ==> Cloudinary does not support transformations on .ai or .svg FML!!!
 * when using `<Transformation crop='scale' />` you must provide either width or height
 * cannot just use `aspect_ratio`
 *
 *
 * QUESTION
 * Integration with the following:
 * https://github.com/stratiformltd/react-loadable-visibility
 * https://github.com/thejameskyle/react-loadable
 *
 * REVIEW
 *
 * Possibly better to generate a "transformed" sprite sheet?
 * One request to Cloudinary for Sprite & matching CSS (2 pieces of data, 1 request)
 * Then I would have to figure out how to pull the images from the sprite and display them.
 */
