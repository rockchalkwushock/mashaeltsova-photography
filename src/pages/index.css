.grid {
  align-content: center;
  display: grid;
  grid-gap: 20px 10px;
  grid-template-columns: repeat(auto-fit, minmax(280px, 1fr));
  justify-items: center;
  padding: 2em 0;
}

.box {
  box-shadow: 2px 2px 20px 1px rgba(0, 0, 0, 0.5);
  height: 240px;
  width: 240px;
}

.one,
.three {
  background-color: lightblue;
}

.two,
.four {
  background-color: lightcoral;
}

@media (min-width: 540px) {
  .grid {
    grid-template-columns: repeat(auto-fit, minmax(340px, 1fr));
  }
  .box {
    width: 400px;
  }
}
@media (min-width: 736px) {
  .box {
    width: 300px;
  }
}
@media (min-width: 1200px) {
  .grid {
    grid-template-columns: repeat(auto-fit, minmax(230px, 1fr));
    grid-gap: 10px;
  }
  .box {
    width: 300px;
  }
}
