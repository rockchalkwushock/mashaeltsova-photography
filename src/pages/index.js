import React from 'react'

import { Title } from '../components'

const IndexPage = () => (
  <div>
    <Title text="Refactor Engage!" />
  </div>
)

export default IndexPage
