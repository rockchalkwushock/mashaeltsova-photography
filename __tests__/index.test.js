import React from 'react'
import { render, shallow } from 'enzyme'
import { intl } from 'react-intl'

/**
 * QUESTION
 *
 * How to toggle the locale & messages with mock?
 */

import { IndexPage } from '../pages'

describe('IndexPage "en"', () => {
  test.skip('should yield English text.', () => {
    const wrapper = shallow(<IndexPage intl={intl} />)
    expect(wrapper.text()).toEqual('Hello World!')
  })

  test('Snapshot', () => {
    const tree = render(<IndexPage intl={intl} />)
    expect(tree).toMatchSnapshot()
  })
})

describe.skip('IndexPage "ru"', () => {
  test('should yield Russian text.', () => {
    const wrapper = shallow(<IndexPage intl={intl} />)
    expect(wrapper.text()).toEqual('Hello World!')
  })

  test('Snapshot', () => {
    const tree = render(<IndexPage intl={intl} />)
    expect(tree).toMatchSnapshot()
  })
})
