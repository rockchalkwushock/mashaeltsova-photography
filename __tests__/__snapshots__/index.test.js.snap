// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage "en" Snapshot 1`] = `
<div>
  <div
    class="jsx-867203339"
  />
  <main
    class="jsx-1548676517 grid"
  >
    <section
      class="jsx-1268207398 header"
    >
      Header
    </section>
    <nav>
      Navigation
    </nav>
    <section
      class="jsx-1268207398 about"
    >
      About
    </section>
    <section
      class="jsx-1268207398 services"
    >
      Services
    </section>
    <section
      class="jsx-1268207398 gallery"
    >
      Gallery
    </section>
    <section
      class="jsx-1268207398 booking"
    >
      Booking
    </section>
    <footer>
      Footer
    </footer>
  </main>
</div>
`;

exports[`IndexPage "ru" Snapshot 1`] = `
<h1>
  Hello World!
</h1>
`;
